hypothesis-python/src/hypothesis/extra/django/_fields.py

```diff
@@ -262,6 +262,51 @@ def _for_form_boolean(field):
     return st.booleans()
 
 
+def _model_choice_strategy(field):
+    def _strategy():
+        if field.choices is None:
+            # The field was instantiated with queryset=None, and not
+            # subsequently updated.
+            raise InvalidArgument(
+                "Cannot create strategy for ModelChoicesField with no choices"
+            )
+        elif hasattr(field, "_choices"):
+            # The choices property was set manually.
+            choices = field._choices
+        else:
+            # choices is not None, and was not set manually, so we
+            # must have a QuerySet.
+            choices = field.queryset
+
+        if not choices.ordered:
+            raise InvalidArgument("QuerySet must have explicit ordering")
```

Comment:
This could probably use a better message, like what's in `check_sample`. The reason I've put this here is that it seems important to check that the queryset has an explicit ordering, but to get the values for `sampled_from` we should probably use the public API of `choices`, as it takes care of the fact that the choices might have been set after instantiation, and adds the empty choice.

Reply:
Added more detail to the error message in e7f018033